src/NuGet.Core/NuGet.Common/UriUtility.cs

```diff
@@ -51,5 +53,31 @@ public static string UrlEncodeOdataParameter(string value)
 
             return value;
         }
+
+        /// <summary>
+        /// Convert a file:// URI to a local path.
+        /// If the input can be parsed this will return Uri.LocalPath, if the input 
+        /// is not a URI or fails to parse the original string will be returned.
+        /// </summary>
+        /// <param name="localOrUriPath">Possible file:// URI path or local path.</param>
+        public static string GetLocalPath(string localOrUriPath)
+        {
+            // check if this starts with file://
+            if (!string.IsNullOrEmpty(localOrUriPath)
+                && localOrUriPath.StartsWith(FilePrefix, StringComparison.OrdinalIgnoreCase))
+            {
+                // convert to a uri and get the local path
+                Uri uri = TryCreateSourceUri(localOrUriPath, UriKind.RelativeOrAbsolute);
```

Comment:
It's the same as calling directly `Uri.TryCreate`. Don't need to `FixSourceUri` in this case.